### PR TITLE
[WIP] refactor: decomposition selection utils

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

https://github.com/ngxs/store/issues/913

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/store/src/utils/selector/selector-processor.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  SharedSelectorOptions,
+  StateClassInternal
+} from '../../internal/internals';
+import {
+  CreationMetadata,
+  RuntimeSelectorInfo,
+  selectorOptionsMetaAccessor
+} from './selector.tokens';
+
+export class SelectorProcessor {
```
Comment (markwhitfeld) [inline packages/store/src/utils/selector/selector-processor.ts]: Please don't make a class. I can see your Java habits coming through here 😆 
These can just be exported functions.

Comment (markwhitfeld) [reply packages/store/src/utils/selector/selector-processor.ts]: In the light of the main comment this file grouping doesn't make sense either.

**Comment by markwhitfeld** [comment]
PS. As a general rule of thumb for myself if I ever fell tempted to call something a `processor`, `manager`, `helper`, `service` or even `controller`(outside of the architectural pattern) and the only other descriptive part is a noun of the thing that it acts on then I probably don't know what it does or it's responsibilities are not clear. Naming is hard, but it can also reveal a lot about what we are building.
Generally named classes like these tend to become god classes and dumping grounds for anything related to the noun.
Just a little tip that I have found useful in my experience.
cc @arturovt

**Comment by splincode** [comment]
@markwhitfeld However, I am not a supporter of dumping functions in one file, can we somehow agree on this?

**Pull request closed without merging**